indicators/Hurst/Hurst.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Skender.Stock.Indicators
+{
+    public static partial class Indicator
+    {
+        // HURST EXPONENT
+        /// <include file='./info.xml' path='indicator/*' />
+        /// 
+        public static IEnumerable<HurstResult> GetHurst<TQuote>(
+            this IEnumerable<TQuote> quotes,
+            int lookbackPeriods = 100)
+            where TQuote : IQuote
+        {
+
+            // sort quotes
+            List<TQuote> quotesList = quotes.Sort();
+
+            // check parameter arguments
+            ValidateHurst(quotes, lookbackPeriods);
+
+            // initialize
+            int size = quotesList.Count;
+            List<HurstResult> results = new(size);
+
+
+            // roll through quotes
+            for (int i = 0; i < size; i++)
+            {
+                int index = i + 1;
+                TQuote q = quotesList[i];
+
+                HurstResult result = new()
+                {
+                    Date = q.Date
+                };
+
+                if (index > lookbackPeriods)
+                {
+                    // get evaluation batch
+                    decimal[] values = new decimal[lookbackPeriods];
+
+                    int x = 0;
+                    for (int p = index - lookbackPeriods; p < index; p++)
+                    {
+                        // compile return values
+                        if (quotesList[p - 1].Close != 0)
+                        {
+                            values[x] = quotesList[p].Close / quotesList[p - 1].Close - 1;
+                        }
+
+                        x++;
+                    }
+
+                    // calculate hurst exponent
+                    result.HurstExponent = CalcHurst(values);
+                }
+                results.Add(result);
+            }
+
+            return results;
+        }
+
+        private static decimal CalcHurst(decimal[] values)
+        {
+            int setNum = 1;
+            int totalSize = values.Length;
+            double[] logRs = new double[6];
+            double[] logSize = new double[6];
+
+            // roll through sets
+            for (int n = 1; n <= 32; n *= 2)
+            {
+                // initialize sets
+                int nominalChunkSize = totalSize / n;
```

Comment:
Is this wise, to add `1`, or should be just omit the first observation?  This may impact cumulative mean deviation calculations.

Reply:
No.  I'm changing this to use consistently sized chunks; though there might be a few unused observations in larger datasets.